api/v1beta1/awsmachine_webhook.go

```diff
@@ -140,6 +144,37 @@ func (r *AWSMachine) validateCloudInitSecret() field.ErrorList {
 	return allErrs
 }
 
+func (r *AWSMachine) cloudInitEnabled() bool {
```

Comment:
It is optional, though it's not a pointer type, so we have to assert all fields. Perhaps it would make sense to rename this function to `cloudInitConfigured`, I think it would make more sense then, as cloudinit can be _enabled_ without configuration and we cannot detect that.

Reply:
cloudInitConfigured() check only covers the scenario where any of cloudInit fields are set and since that is not a requirement for cloudInit, I think we don't need to have a separate logic that covers only part of the case.

Reply:
Hm, I'm not sure what is the issue here. This function is just a name for the block of code used in `validateIgnitionAndCloudInit()`, which verifies that Ignition and cloud-init specific fields are not set at the same time, as it would make either one or another no-op. As far as I understand, user can specify `r.Spec.CloudInit.InsecureSkipSecretsManager` and `r.Spec.CloudInit.SecureSecretsBackend`, which would indicate that they expect cloud-init specific behavior. If also Ignition-specific behavior is requested, then the configuration becomes invalid, as both cannot be used at the same time.

I think this function provides value and should stay. I may alter the implementation or inline it if you think this is better, though generally function names should be preferred over comments.